source/compilation/builtins/CoverageFuncs.cpp

```diff
@@ -0,0 +1,107 @@
+//------------------------------------------------------------------------------
+// CoverageFuncs.cpp
+// Coverage control functions
+//
+// File is under the MIT license; see LICENSE for details
+//------------------------------------------------------------------------------
+#include "slang/binding/MiscExpressions.h"
+#include "slang/binding/SystemSubroutine.h"
+#include "slang/compilation/Compilation.h"
+#include "slang/diagnostics/SysFuncsDiags.h"
+#include "slang/symbols/InstanceSymbols.h"
+#include "slang/syntax/AllSyntax.h"
+
+namespace slang::Builtins {
+
+class CoverageNameOrHierFunc : public SystemSubroutine {
+public:
+
+    CoverageNameOrHierFunc(const std::string& name, const Type& returnType, unsigned int nameOrHierIndex,
+                           size_t requiredArgs = 0, const std::vector<const Type*>& argTypes = {}) :
+        SystemSubroutine(name, SubroutineKind::Function),
+        argTypes(argTypes), returnType(&returnType),
+        nameOrHierIndex(nameOrHierIndex), requiredArgs(requiredArgs) {
+        ASSERT(requiredArgs <= argTypes.size());
+        ASSERT(nameOrHierIndex <= argTypes.size());
+    };
+
+    virtual const Expression& bindArgument(size_t argIndex, const BindContext& context,
+                                           const ExpressionSyntax& syntax, const Args& args) const final {
+        const BindContext* ctx = &context;
```

Comment:
This variable isn't really doing anything is it?

Reply:
Copy-paste :-)